### DISCUSSION TITLE
Entity Listener

### DISCUSSION THREAD

**Discussion by gtsluk** [MorphiaOrg/morphia#2385] [General]
```java
@Entity(value = "entity", useDiscriminator = false)
@EntityListeners(FieldSigner.class)
public class Entity Aextends RootDocument {
    private String name;
    @EncryptedField
    private String businessEmail;
}

@Target({ElementType.FIELD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface EncryptedField {
}

@Singleton
@Startup
public class FieldSigner {
    @PrePersist
    void prePersist(Object entity) throws IllegalAccessException {}
    @PostLoad
    void onPostLoad( Object entity ) throws IllegalAccessException {
}
```
The signer class may be triggered when the entity register in morphia lifeCycle decoder. Are there any properties ensure the entity class being mapped in the lifeCycleDeoced?

**Comment by evanchooly** [comment]
I'm not sure I understand your question.  Could you elaborate?

**Reply by gtsluk** [reply]
I have several Mongodb collections with relevant POJO entities. I initialize the Morphia session without mapping any class. I implemented an externalized class performing persist and post-load action. I found that the post-load or persis action may be triggered randomly after the deployment with Quarkus V2.9 on the open-shift platform. Is there any setting in Morphia to ensure the action should be started?

**Reply by evanchooly** [reply]
You should map your entities explicitly even if by just passing the package name in.  (There's a property in the config for quarkus-morphia for that that takes a comma-delimited list of package names.)  Morphia *can* map entities on the fly but without prior knowledge of mapping data, some things might not be (immediately) available.

**Reply by gtsluk** [reply]
do you mean I call morphia.mapPackage() or 
[quarkus.morphia.packages] in the application.properties? (https://github.com/MorphiaOrg/morphia/discussions/2385#quarkus-morphia_quarkus.morphia.packages)

**Reply by gtsluk** [reply]
we're using 
       <dependency>
            <groupId>dev.morphia.morphia</groupId>
            <artifactId>morphia-core</artifactId>
            <version>2.2.6</version>
        </dependency>

**Reply by evanchooly** [reply]
If you're using quarkus then configure it via `application.properties`.

**Reply by gtsluk** [reply]
we use the Morphia core library rather than QUARKUS-MORPHIA. Therefore, the solution is stored the package name in the form of comma-separated in the application properties and called morphia.mapPackage() after creating the morphia. Please correct me if I moved in the wrong direction.

**Reply by evanchooly** [reply]
So is this a quarkus app or no?  It feels like we've gone back and forth on the point.  If it's a quarkus app, you really should use `quarkus-morphia` and `application.properties` to make sure you take advantage of the work that the extension will (eventually) do for you.  But if you list the packages in `application.properties`, there is no need to call `mapPackage()` as the extension will do that for you.

On the other hand, there's nothing wrong with using Morphia directly.  You just won't get any build time support and probably native mode will break on you if that's a thing you care about.  Not everyone does.

**Reply by gtsluk** [reply]
It is a Quarkus app. But I found the eventListener eventually works on OpenShift.
